src/Tizen.Multimedia/AudioManager/AudioStreamPolicy.cs

```diff
@@ -304,6 +304,72 @@ public void RemoveDeviceForStreamRouting(AudioDevice device)
                 ThrowIfError("Failed to remove device for stream routing");
         }
 
+        /// <summary>
+        /// Gets or sets the preferred input device id.
+        /// </summary>
+        /// <value>
+        /// A device id which is greater than or equal to 0.<br/>
+        /// The default is 0 which means any device is not set on this property.
+        /// </value>
+        /// <remarks>
+        /// This property is to set a specific built-in device when the system has multiple devices of the same built-in device type.
+        /// When there's only one device for a built-in device type in the system, nothing will happen even if this property is set successfully.
+        /// </remarks>
+        /// <exception cref="ArgumentException">A device is not for input.</exception>
+        /// <exception cref="InvalidOperationException">An internal error occurs.</exception>
+        /// <exception cref="AudioPolicyException">A device is not supported by this <see cref="AudioStreamPolicy"/> instance.</exception>
+        /// <exception cref="ObjectDisposedException">The <see cref="AudioStreamPolicy"/> has already been disposed of.</exception>
+        /// <seealso cref="AudioManager.GetConnectedDevices()"/>
+        /// <since_tizen> 6 </since_tizen>
+        public int PreferredInputDeviceId
```

Comment:
Is it a `AudioDevice`?
How about use `AudioDevice` type instead of int?
because, App developer can't know about Id, app developer should get a AudioDevice first.

and others API also use `AudioDevice` 
` public void RemoveDeviceForStreamRouting(AudioDevice device)`

```suggestion
        public AudioDevice PreferredInputDevice
```

Reply:
Here' a reason why I use integer value instead of the AudioDevice.

AudioDevice instance can get from new List() via inside of GetConnectedDevices() as an item. Hence, if we use AudioDevice as the return value, the app developer should not call FreeDeviceList() between setting PreferredInput[Output]Device property and getting PreferredInput[Output]Device.

Device id of integer value, on the other hand, doesn't need to care about the lifecycle of AudioDevice. What the app. developer has to care about it is to get new AudioDevice instance and check the id one of its property(audioDevice.Id).

In a situation like this, I'd like to adopt the latter.

Reply:
Does not matter call FreeDeviceList(), 
`FreeDeviceList` is just release a native list object, and a AudioDevice object is created on this
https://github.com/Samsung/TizenFX/blob/7d5b33bd6d146e0504a4398f9b97b44a85718156/src/Tizen.Multimedia/AudioManager/AudioManager.cs#L82
and this lifecycle is managed by AudioDevice

Reply:
I checked again you're right that there's no FreeDeviceList() for CSharp API, but just for interal operation.

I'll revise it with doing like below.
 Set the property with handle -> invoke interop. setter method with audioDevice.Id and save the instance.
 Get the property with handle -> check the id from the saved instance with interop. getter method. and return the saved instance.